src/main/java/org/sonarqube/gradle/SonarQubePlugin.java

```diff
@@ -92,7 +92,7 @@ private void configureTask(SonarQubeTask sonarQubeTask, Project project, Map<Str
       .filter(p -> p.getPlugins().hasPlugin(JavaPlugin.class) && !p.getExtensions().getByType(SonarQubeExtension.class).isSkipProject())
       .map(p -> p.getTasks().getByName(JavaPlugin.TEST_TASK_NAME))
       .collect(Collectors.toList());
-    sonarQubeTask.dependsOn(testTasks);
+    sonarQubeTask.mustRunAfter(testTasks);
```

Comment:
I wonder if we should not add a `dependsOn` on the JavaPlugin.COMPILE_JAVA_TASK_NAME/COMPILE_JAVA_TEST_TASK_NAME because contrary to tests, I see no use case where you don't want it have bytecode generated prior to the SQ analysis. It was previously implicit due to the transitive dependency sonarqube -> JavaPlugin.TEST_TASK_NAME -> (COMPILE_JAVA_TASK_NAME, COMPILE_JAVA_TEST_TASK_NAME)

Reply:
good point!